src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -2,18 +2,125 @@
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
 
+    private static final int INITIAL_CAPACITY = 16;
+    private static final double INITIAL_LOAD_FACTOR = 0.75;
+    private Entry<K, V>[] table = new Entry[INITIAL_CAPACITY];
+    private int size = 0;
+
     @Override
     public void put(K key, V value) {
-
+        if (size >= INITIAL_LOAD_FACTOR * table.length) {
+            resizing();
+        }
+        if (key == null) {
+            putForNullKey(value);
+            return;
+        }
+        int hashCode = key.hashCode();
+        int index = (table.length - 1) & hashCode;
+        Entry<K, V> entry = table[index];
+        if (entry == null) {
+            table[index] = new Entry<>(key, value, null);
+            size++;
+            return;
+        }
+        Entry<K, V> currentEntry = entry;
+        while (true) {
+            if (key.equals(currentEntry.key)) {
+                currentEntry.value = value;
+                return;
+            }
+            if (currentEntry.next == null) {
+                currentEntry.next = new Entry<>(key, value, null);
+                size++;
+                return;
+            }
+            currentEntry = currentEntry.next;
+        }
     }
 
     @Override
     public V getValue(K key) {
+        if (size == 0) {
+            return null;
+        }
+        if (key == null) {
+            return getForNullKey();
+        }
+        int hashCode = key.hashCode();
+        int index = (table.length - 1) & hashCode;
+        Entry<K, V> entry = table[index];
+        while (entry != null) {
+            if (key.equals(entry.key)) {
+                return entry.value;
+            }
+            entry = entry.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private void putForNullKey(V value) {
+        Entry<K, V> entry = table[0];
+        if (entry == null) {
+            table[0] = new Entry<>(null, value, null);
```

Comment:
Diamond operator. Be attentive with the diamond operator in java. What is the difference between: `new Node<>(...)` and `new Node<K,V>(...)`and `new Node(...)`. Read more about it [here](https://www.baeldung.com/java-diamond-operator)